# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/EarlyHintPreloader.h
# Commit: 1177913e1edf
# Full Hash: 1177913e1edf443da73d438f95a334e466b623e2
# Author: Manuel Bucher <manuel@mozilla.com>
# Date: 2022-12-02 21:26:33
# Regressor Bug: 1771867
# File Overlap Count: 1
# Description:
#   Bug 1771867 - Early Hints Phase 2 - Part 8: Add list of open channels from OngoingEarlyHints to RedirectToRealChannelArgs r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161179
# ==============================================================================

diff -r d33ccbbf407d -r 1177913e1edf netwerk/protocol/http/EarlyHintPreloader.h
--- a/netwerk/protocol/http/EarlyHintPreloader.h	Fri Dec 02 09:45:26 2022 +0000
+++ b/netwerk/protocol/http/EarlyHintPreloader.h	Fri Dec 02 09:45:27 2022 +0000
@@ -22,6 +22,7 @@
 namespace mozilla::net {
 
 class EarlyHintPreloader;
+class EarlyHintConnectArgs;
 struct LinkHeader;
 
 // class keeping track of all ongoing early hints
@@ -38,9 +39,17 @@
 
   void CancelAllOngoingPreloads();
 
+  // registers all channels and returns the ids
+  void RegisterLinksAndGetConnectArgs(
+      nsTArray<EarlyHintConnectArgs>& aOutLinks);
+
  private:
   ~OngoingEarlyHints() = default;
   nsRefPtrHashtable<PreloadHashKey, EarlyHintPreloader> mOngoingPreloads;
+
+  // keep track of all preloads in the order they were specified in the early
+  // hint header
+  nsTArray<EarlyHintConnectArgs> mLinks;
 };
 
 class EarlyHintPreloader final : public nsIStreamListener,